### PR TITLE
DDCYLS-170

### PR DESCRIPTION
**Pull request by dips7189**
- Integration tests now live in the "it" directory underneath the test directory.
- Unit tests live in "unit" directory underneath test.

**To run the unit tests:** `sbt test`

**To run the Integration tests:**

- `sbt it:test`
- `sbt IntegrationTest/test`

**Other changes**
- `macwire` has been completely abolished
- the `build.sbt` file has gone through quite a bit of refactoring.
- `WiremockSpec` has been renamed to `WiremockServer` ... because it isn't a test class
- `FakeApplicationSpec` has been renamed to `FakeApplicationSetup` because it isn't a test class
- `TestWiring` has been deleted since it contained setup for the application that really belonged in `FakeApplicationSetup`

### CODE REVIEW

File: test/unit/uk/gov/hmrc/digitalservicestaxfrontend/TestInstances.scala [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+    RegexpGen.from(BuildingSocietyRollNumber.regex).map(BuildingSocietyRollNumber.apply).retryUntil(x => x == x.trim)
+  }
+
+  implicit val arbSortCode: Arbitrary[SortCode] = Arbitrary {
+    RegexpGen.from(SortCode.regex).map(SortCode.apply)
+  }
+
+  implicit val argRestrictedString: Arbitrary[RestrictiveString] = Arbitrary(
+    RestrictiveString.gen
+    //    RegexpGen.from("""^[0-9a-zA-Z{À-˿’}\\- &`'^._|]{1,255}$""")
```
Comment (PatrykRudnicki) [inline test/unit/uk/gov/hmrc/digitalservicestaxfrontend/TestInstances.scala:171]: Is this comment required?

**Review by PatrykRudnicki — APPROVED** [review]
LGTM! Great job!

File: test/unit/uk/gov/hmrc/digitalservicestaxfrontend/TestInstances.scala (hunk 1 above)
Comment (PatrykRudnicki) [reply test/unit/uk/gov/hmrc/digitalservicestaxfrontend/TestInstances.scala:171]: The comment that you mentioned I think we should keep them

**Review by chandaksweta — APPROVED** [review]

File: test/unit/uk/gov/hmrc/digitalservicestaxfrontend/TestInstances.scala (hunk 1 above)
Comment (hmrclt) [reply test/unit/uk/gov/hmrc/digitalservicestaxfrontend/TestInstances.scala:171]: The whole issue with RFC compliant emails is, I think, estate-wide and I've never heard a satisfactory solution. 
We could validate against the RFC, but then we would be admitting email addresses that HoD's would almost certainly choke upon.

Comment (dips7189) [reply test/unit/uk/gov/hmrc/digitalservicestaxfrontend/TestInstances.scala:171]: Interesting, I'll take a look into this in the mean time. Sounds quite interesting.

**Pull request merged**